src/main/java/uk/gov/companieshouse/api/accounts/controller/PreviousPeriodController.java

```diff
@@ -37,24 +44,45 @@
     @Autowired
     private ApiResponseMapper apiResponseMapper;
 
+    @Autowired
+    private ErrorMapper errorMapper;
+
+    @Autowired
+    private PreviousPeriodValidator previousPeriodValidator;
+
     @PostMapping
-    public ResponseEntity create(@Valid @RequestBody PreviousPeriod previousPeriod,
-        HttpServletRequest request) {
+    public ResponseEntity create(@Valid @RequestBody PreviousPeriod previousPeriod, BindingResult bindingResult,
+        @PathVariable("companyAccountId") String companyAccountId, HttpServletRequest request) {
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult, errors);
+
+        }
+
+        previousPeriodValidator.validatePreviousPeriod(previousPeriod, errors);
```

Comment:
Should we be performing validation if there are binding errors? This could open us up to NPE's if a non-null field were actually null.

Reply:
The validator just validates the total field calculations, and performs null checks in the method.